oida/commands/componentize.py

```diff
@@ -185,3 +196,140 @@ class {class_name}(AppConfig):
             AppConfigUpdater(class_name, new_module, default_app_label)
         )
         apps_py_path.write_text(run_black(updated_module.code))
+
+
+class CeleryTaskNameUpdater(ContextAwareTransformer):
+    """
+    This updater searches for Celery tasks defined with the @app.task decorator, where
+    the name of the task is implicitly set. This name includes the folder structure of the Django app.
+    Because we need to keep the name the same if we move apps into components, we need to set the name
+    explicitly to the original name, i.e. without the component name.
+    It also checks whether the name has already been explicitly set, if so the name is NOT changed.
+    """
+
+    def __init__(
+        self, context: CodemodContext, old_module: str, new_module: str
+    ) -> None:
+        super().__init__(context=context)
+        self.old_module = old_module
+        self.new_module = new_module
+        self.in_app: int = 0
+        self.function_name: str | None = None
+
+    def transform_module_impl(self, tree: cst.Module) -> cst.Module:
+        if self.context.full_module_name is not None:
+            current_module_name_list = self.context.full_module_name.split(".")
+            new_module_name_list = self.new_module.split(".")
+            # The full module name is in the form: product_availability.core.services.data.supply
+            # while the module we are changing is of the form: tienda.product_availability.core
+            # We, therefore, need to check if the last two (string) components of the name of the
+            # new module are equal to the first two (string) components of the name of the module
+            # currently being processed. If we do not do this test - all tasks in the project
+            # (e.g. tienda) will be prepended with the component name, which is wrong.
```

Comment:
Would probably be good to generalize Tienda/Oda-specific things :)

Reply:
I changed some of the text in the comment, hope you agree!